File: GLaDOS_Checkin.js
const axios = require('axios');
const COOKIES = process.env['GLaDOS_Cookie'];   // 需要的环境变量


const Bytes2GB = 1 * 1024 * 1024 * 1024;
const checkIn = async (cookie) => {
    return axios({
        method: 'post',
        url: 'https://glados.rocks/api/user/checkin',
        headers: {
            'Cookie': cookie,
            'origin': 'https://glados.rocks',
            'user-agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.4896.127 Safari/537.36',
            'content-type': 'application/json;charset=UTF-8',
            'content-length': 26
        },
        data: {
            token: "glados.network"
        }
    });
};

const getStatus = async (cookie) => {
    return axios({
        method: 'get',
        url: 'https://glados.rocks/api/user/status',
        headers: {
            'Cookie': cookie
        }
    });
};

const checkInAndGetStatus = async (cookie) => {
    const checkInMessage = (await checkIn(cookie))?.data?.message;

    const userStatus = (await getStatus(cookie))?.data?.data;
    const email = userStatus?.email;
    const leftDays = parseInt(userStatus?.leftDays);
    const traffic = (parseInt(userStatus?.traffic) / Bytes2GB).toFixed(2);
    const usedDays = parseInt(userStatus?.days);
    return {
        '账号': email,
        '天数': leftDays,
        '流量': traffic,
        '已用天数': usedDays,
        '签到情况': checkInMessage
    };
};

const pushplus = (token, infos) => {
    const titleEmail = infos?.[0]['账号'];
    const titleLeftDays = infos?.[0]['天数'];
    const titleCheckInMessage = infos?.[0]['签到情况'];
    const titleSpace = 4;

    const title = (
        '账号: ' + `${titleEmail}`.padEnd(titleEmail.length + titleSpace) +
        '天数: ' + `${titleLeftDays}`.padEnd(titleLeftDays.toString().length + titleSpace) +
        '签到情况: ' + `${titleCheckInMessage}`
    ).slice(0, 100);

    const data = {
        token,
        title,
        content: JSON.stringify(infos),
        template: 'json'
    };
    console.log(data);

    return axios({
        method: 'post',
        url: `http://www.pushplus.plus/send`,
        data
    });
};


const printLog = (infos) => {
    const contentEmail = infos?.[0]['账号'];
    const contentLeftDays = infos?.[0]['天数'];
    const contentCheckInMessage = infos?.[0]['签到情况'];
    const contentTraffic = infos?.[0]['流量'];
    const contentUsedDays = infos?.[0]['已用天数'];
    const contentLine = '\n';
    const titleSpace = 4;

    const content = (
        '账号: ' + `${contentEmail}`.padEnd(contentEmail.length + titleSpace) + contentLine +
        '天数: ' + `${contentLeftDays}`.padEnd(contentLeftDays.toString().length + titleSpace) + contentLine +
        '流量: ' + `${contentTraffic}`.padEnd(contentTraffic.length) + 'GB'.padEnd('GB'.length + titleSpace) + contentLine +
        '已用天数: ' + `${contentUsedDays}`.padEnd(contentUsedDays.toString().length + titleSpace) + contentLine + 
        '签到情况: ' + `${contentCheckInMessage}`
    ).slice(0, 100);

    return content;
}
const GLaDOSCheckIn = async () => {
    try {
        console.log('GLaDOS，开始签到：')
        // const cookies = process.env.COOKIES?.split('&&') ?? [];
        const cookies = COOKIES.split('&&') ?? [];
        const infos = await Promise.all(cookies.map(async cookie => await checkInAndGetStatus(cookie)));
        console.log(printLog(infos));

        // const PUSHPLUS = process.env.PUSHPLUS;

        // if (PUSHPLUS && infos.length) {
        //     const pushResult = (await pushplus(PUSHPLUS, infos))?.data?.msg;
        //     console.log(pushResult);
        // }
    } catch (error) {
        console.log(error);
    }
};

GLaDOSCheckIn();

File: GLaDOS_Checkin.py
# -*- coding: utf-8 -*-
# ==============================================================================
# @ Author :
# @ Desc : 
# @ Date : 2022/7/24
# ==============================================================================
import json
import math
import os

import requests

GET_STATUS_URL = "https://glados.rocks/api/user/status"
CHECK_IN_URL = "https://glados.rocks/api/user/checkin"
COOKIE_STR = os.environ.get('GLaDOS_Cookie')
QYWX_KEY = os.environ.get('GLaDOS_Cookie')
GLaDOS_SEND = os.environ.get('GLaDOS_SEND')


class GladosCheckIn:

    def __init__(self):
        self.headers = {
            "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 \
            (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36",
        }
        self.cookies = {cookie.split("=")[0]: cookie.split("=")[1] for cookie in COOKIE_STR.split(";")}
        self.res = {}

    def get_status(self):

        resp = requests.get(url=GET_STATUS_URL, headers=self.headers, cookies=self.cookies)
        resp.raise_for_status()

        data = self.resp_json(resp)
        if data["code"] != 0:
            raise Exception("get status error")

        self.res["email"] = data["data"]["email"]
        self.res["left_days"] = float(data["data"]["leftDays"])
        self.res["traffic"] = self.convert_traffic_size(int(data["data"]["traffic"]))
        self.res["used_days"] = data["data"]["days"]

        return self

    def check_in(self):

        data = {"token": "glados.network"}

        resp = requests.post(url=CHECK_IN_URL, data=data, headers=self.headers, cookies=self.cookies)
        resp.raise_for_status()

        data = self.resp_json(resp)
        self.res["message"] = data.get("message")

        return self

    def print_result(self):
        print("账号: {}".format(self.res["email"]))
        print("剩余天数: {}".format(self.res["left_days"]))
        print("流量: {}".format(self.res["traffic"]))
        print("已用天数: {}".format(self.res["used_days"]))
        print("签到情况: {}".format(self.res["message"]))
        self.wechat(self.res)

    @staticmethod
    def resp_json(resp):
        encoding = requests.utils.guess_json_utf(resp.content)
        if encoding is not None:
            try:
                return json.loads(resp.content.decode(encoding))
            except UnicodeDecodeError:
                pass
        return json.loads(resp.text)

    @staticmethod
    def convert_traffic_size(size_bytes):
        if size_bytes == 0:
            return "0B"
        size_name = ("B", "KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB")
        i = int(math.floor(math.log(size_bytes, 1024)))
        p = math.pow(1024, i)
        s = round(size_bytes / p, 2)
        return "%s %s" % (s, size_name[i])

    @staticmethod
    def wechat(res):
        if QYWX_KEY and GLaDOS_SEND:
            url = f'https://qyapi.weixin.qq.com/cgi-bin/webhook/send?key={QYWX_KEY}'
            text = 'GlaDOS签到详情'
            desp = "账号: {}\n剩余天数: {}\n流量: {}\n已用天数: {}\n签到情况: {}".format(
                res["email"],
                res["left_days"],
                res["traffic"],
                res["used_days"],
                res["message"]
            )
            json_data = {
                'msgtype': 'text',
                'text': {
                    'content': f'{text}\n\n{desp}',
                }
            }
            headers = {
                'Content-Type': 'application/json'
            }
            rst = requests.post(url, json=json_data, headers=headers).text
            print(rst)


if __name__ == '__main__':
    checkin = GladosCheckIn()
    checkin.check_in().get_status().print_result()
